### ISSUE TITLE
Star Wars: Dark Forces (Steam ID: 32400)

### ISSUE THREAD

**Issue by Rashkavar** [dreamer/boxtron#39]
## Compatibility Report
- Link to Steam store: https://store.steampowered.com/app/32400/STAR_WARS__Dark_Forces/
- Link to GOG store:  
- Link to ProtonDB: https://www.protondb.com/app/32400

### System Information
- Linux distribution: Mint 19.3
- Kernel: 5.3.0-45-generic
- Desktop: Cinnamon
- DOSBox: 0.74-4.3

### Report
- The game runs out of the box: Yes

### Configuration changes
None

### Report
Game runs well, but no background music plays.  Can get background music using proton to run the Windows version of dosbox, but this results in frame rate issues that make it unplayable - these issues are completely solved running through boxtron.

Is there enough crossover between Windows and Linux versions of dosbox that I could find whatever configuration is made on packaged Windows version and copy it over to the dosbox being used through this?  If so, where should I look for the relevant config info?

**Comment by dreamer** [comment]
About music no playing: did you follow Boxtron installation instructions and do you have software MIDI synthesiser installed? Either Timidity with FluidR3 soundfont or Fluidsynth with FluidR3 soundfont?

**Comment by Rashkavar** [comment]
Seems you pointed me in the right direction: I'd followed the directions and had all dependencies installed, but it seems Timidity is, by default, set up to use something called freepats.  From /etc/timidity/timidity.cfg - the last several lines:

# By default, try to use the instrument patches from freepats:
source /etc/timidity/freepats.cfg

# alternatively, you can use the fluid-soundfont:
# source /etc/timidity/fluidr3_gm.cfg
# source /etc/timidity/fluidr3_gs.cfg

Just switch the comments around so it uses fluidr3_gm.cfg and fluidr3_gs.cfg instead of freepats.cfg, and it worked.  (After a system reboot - the guide I found on configuring timidity <https://www.rosegardenmusic.com/wiki/setting_up_the_fluidr3_gm.sf2_for_timidity> had a thing about re-initiallizing timidity that I couldn't get to work and I figured a reboot would force timidity to re-initialize as well.  I didn't test without doing the reboot because I'm still so used to Windows that rebooting after changing settings is basically reflex.)

**Comment by dreamer** [comment]
Out of curiosity, which distribution sets up such Timidity configuration by default? Maybe we should still treat this as a bug… perhaps code responsible for starting Timidity could be more robust.

**Comment by Rashkavar** [comment]
Linux Mint 19.3 64-bit.  (Using Cinnamon, if it matters for some reason.)  Maybe there's some way to force Timidity to use FluidR3 configs instead of its normal settings for this?  Though that may be a ham-fisted approach to things.

Did run into some issues with a CTD after playing in a mission for a while with the music working, but reinstalling boxtron fixed the problem.  I did a fair bit of tinkering with boxtron.conf to see if I could get things sorted from there; I'm guessing I broke something on my end.  (I should *really* get into the habit of making backups of config files rather than just assuming I'll remember what changes I made...)

Thanks for the quick feedback - I wouldn't have thought to look at Timidity's settings if not for your wording of "Timidity with Fluid3R."  I knew I had both, because I'd used the apt-get command you'd provided (MInt being in the Debian family), but didn't realize they were two parts of a single system.
